### PR TITLE
Adapt `provider-local` to use priority class `gardener-system-900`

### PR DESCRIPTION
**Pull request by krgostev**
**How to categorize this PR?**

/area control-plane
/kind enhancement

**What this PR does / why we need it**:
This PR removes the custom priority class from the provider-local charts and adapts it to use priorityClass `gardener-system-900` as part of https://github.com/gardener/gardener/issues/5634
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by acumino — COMMENTED** [review]
`provider-local` mcm should also be adapted to use priority class `gardener-system-300`.

**Comment by rfranzke** [comment]
Also, the local path provisioner (https://github.com/gardener/gardener/tree/master/charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates) should get the right priority class

**Comment by krgostev** [comment]
@rfranzke, @timebertt 
> Also, the local path provisioner (https://github.com/gardener/gardener/tree/master/charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates) should get the right priority class

Local path provisioner is not described [in the priority classes table](https://github.com/gardener/gardener/blob/master/docs/development/priority-classes.md) What value should we assign to it?

**Comment by rfranzke** [comment]
It is comparable to the CSI driver/controllers

**Review by acumino — COMMENTED** [review]
/lgtm

File: charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/deployment.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ spec:
       labels:
         app: local-path-provisioner
     spec:
+      priorityClassName: gardener-system-200
```
Comment (rfranzke) [inline charts/gardener/provider-local/internal/shoot-system-components/charts/local-path-provisioner/templates/deployment.yaml]: How does this work? This component is running in the shoot and here we should only see `gardener-shoot-system-*` priority classes. 
In this light, `system-cluster-critical` is more appropriate (CSI drivers are using `system-node-critical` but those are `DaemonSet`s)

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**